stcgal/protocols.py

```diff
@@ -1195,50 +1227,62 @@ class Stc12Protocol(StcBaseProtocol):
-        
-        Reads a packet of data from the MCU and and do
-        validity and checksum checks on it.
+    # def read_packet(self):
+    #     """Read and check packet from MCU.
 
-        Returns packet payload or None in case of an error.
-        """
+    #     Reads a packet of data from the MCU and and do
+    #     validity and checksum checks on it.
 
-        # read and check frame start magic
-        packet = bytes()
-        packet += self.read_bytes_safe(1)
-        if packet[0] != self.PACKET_START[0]:
-            raise StcFramingException("incorrect frame start")
-        packet += self.read_bytes_safe(1)
-        if packet[1] != self.PACKET_START[1]:
-            raise StcFramingException("incorrect frame start")
+    #     Returns packet payload or None in case of an error.
+    #     """
 
-        # read direction and length
-        packet += self.read_bytes_safe(3)
-        if packet[2] != self.PACKET_MCU[0]:
-            self.dump_packet(packet)
-            raise StcFramingException("incorrect packet direction magic")
+    #     # read and check frame start magic
+    #     packet = bytes()
+    #     packet += self.read_bytes_safe(1)
+    #     if packet[0] != self.PACKET_START[0]:
+    #         raise StcFramingException("incorrect frame start")
+    #     packet += self.read_bytes_safe(1)
+    #     if packet[1] != self.PACKET_START[1]:
+    #         raise StcFramingException("incorrect frame start")
 
-        # read packet data
-        packet_len, = struct.unpack(">H", packet[3:5])
-        packet += self.read_bytes_safe(packet_len - 3)
+    #     # read direction and length
+    #     packet += self.read_bytes_safe(3)
+    #     if packet[2] != self.PACKET_MCU[0]:
+    #         self.dump_packet(packet)
+    #         raise StcFramingException("incorrect packet direction magic")
 
-        # verify end code
-        if packet[packet_len+1] != self.PACKET_END[0]:
-            self.dump_packet(packet)
-            raise StcFramingException("incorrect frame end")
+    #     # read packet data
+    #     packet_len, = struct.unpack(">H", packet[3:5])
+    #     packet += self.read_bytes_safe(packet_len - 3)
+
+    #     # verify end code
+    #     if packet[packet_len+1] != self.PACKET_END[0]:
+    #         self.dump_packet(packet)
+    #         raise StcFramingException("incorrect frame end")
+
+    #     # verify checksum
+    #     packet_csum, = struct.unpack(">H", packet[packet_len-1:packet_len+1])
+    #     calc_csum = sum(packet[2:packet_len-1]) & 0xffff
+    #     if packet_csum != calc_csum:
+    #         self.dump_packet(packet)
+    #         raise StcFramingException("packet checksum mismatch")
+
+    #     self.dump_packet(packet, receive=True)
 
-        # verify checksum
-        packet_csum, = struct.unpack(">H", packet[packet_len-1:packet_len+1])
-        calc_csum = sum(packet[2:packet_len-1]) & 0xffff
+    #     # payload only is returned
+    #     return packet[5:packet_len-1]
```

Comment:
I left this part commented rather than removed because I want to get some input from you. Method `StcBaseProtocol.read_packet()` needs to cover the case where the start symbol (0x49,0xB6) is missing in some STC89 MCUs, while `read_packet()` in newer protocols do not need to worry about that.  However, I don't think it worths the trouble to maintain a different version of `read_packet()` for those newer protocols, as `StcBaseProtocol.read_packet()` can already handle the normal case.  So I would prefer to keep only the base version.